app/src/main/java/com/sun_asterisk/music_65/screen/playsong/adapter/PlayListAdapter.java

```diff
@@ -0,0 +1,88 @@
+package com.sun_asterisk.music_65.screen.playsong.adapter;
+
+import android.support.annotation.NonNull;
+import android.support.v7.widget.RecyclerView;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.TextView;
+import com.sun_asterisk.music_65.R;
+import com.sun_asterisk.music_65.data.model.Song;
+import com.sun_asterisk.music_65.utils.OnItemRecyclerViewClickListener;
+import java.util.ArrayList;
+import java.util.List;
+
+public class PlayListAdapter extends RecyclerView.Adapter<PlayListAdapter.ViewHolder> {
+
+    private static final int POSITION_START = 0;
+    private List<Song> mSongs;
+    private OnItemRecyclerViewClickListener<Song> mListener;
+
+    public PlayListAdapter() {
+        mSongs = new ArrayList<>();
+    }
+
+    public void setListener(OnItemRecyclerViewClickListener<Song> listener) {
+        mListener = listener;
+    }
+
+    public void updateData(List<Song> songs) {
+        mSongs.clear();
+        mSongs.addAll(songs);
+        notifyItemRangeChanged(POSITION_START, songs.size());
+    }
+
+    public void add(Song song) {
+        mSongs.add(song);
+        notifyItemInserted(getItemCount() - 1);
+    }
+
+    public void loadMoreData(List<Song> songs) {
+        for (Song song : songs)
+            add(song);
```

Comment:
đoạn code "for" này có chức năng gì a?

Reply:
để sau này dùng trong recyclerview tránh bị leak memory khi có quá nhiều dữ liệu

Reply:
a sử dụng như thế nào mà gây leak memory?